common.py

```diff
@@ -6,6 +6,7 @@
 import os.path
 import re
 from datetime import datetime
+from os import makedirs, path
```

Comment:
The `os.path` import was already there... Could you replace usages of `os.path` with `path` instead, and remove the `o.spath` import?